aws/run-on-ec2.py

```diff
@@ -137,17 +146,30 @@ def getPowermixingCommands(runid, s3Manager, instanceIds, instanceIps):
     setupCommands = []
     for i, instanceId in enumerate(instanceIds):
         commands = [
-            "sudo docker pull %s" % (AwsConfig.DOCKER_IMAGE_PATH),
             "mkdir -p config",
             "cd config; curl -sSO %s" % (instanceConfigUrl),
             "mkdir -p sharedata",
+            "git clone https://github.com/lu562/upload-script.git",
+            "cp upload-script/Download_input.sh sharedata/Download_input.sh ",
             "mkdir -p benchmark",
+            "ulimit -n 10000",
         ]
+        threads = []
         for j in range(k):
             fname = f"{powersPrefix}_{runid}_{j}-{i}.share"
-            url = s3Manager.uploadFile(fname)
-            commands.append(f"cd sharedata; curl -sSO {url}")
+            threads.append(threading.Thread(target=uploadFile, args=(fname,)))
```

Comment:
You can start the thread right here instead of iterating over the threads again and starting them. Then, add them to threads list.